src/main/java/io/jsondb/crypto/Default1Cipher.java

```diff
@@ -0,0 +1,166 @@
+  private static final int IV_SIZE = 12;
+  
+  /* intentionally letting the system pick a sane SecureRandom; SecureRandom.getInstanceStrong() can block and is overkill. */
+  private static final SecureRandom rnd = createNewSecureRandom();
+  
+  private static final SecureRandom createNewSecureRandom() {
+    try {
+      return SecureRandom.getInstance("NativePRNGNonBlocking");
+    } catch (NoSuchAlgorithmException e) {
+      try {
+        return SecureRandom.getInstance("SHA1PRNG");
+      } catch (NoSuchAlgorithmException e2) {
+        return new SecureRandom();
+      }
+    }
+  }
+
+  private final Charset charset;
+  private final SecretKeySpec key;
+
+  /**
+   * Creates a new default cipher using 'UTF-8' encoding, with a base64-encoded key.
+   * 
+   * @param base64CodedEncryptionKey  A base 64 encoded symmetric key to be used during encryption and decryption.
+   */
+  public Default1Cipher(String base64CodedEncryptionKey) throws GeneralSecurityException {
+    this(Base64.getDecoder().decode(base64CodedEncryptionKey), StandardCharsets.UTF_8);
+  }
+
+  /**
+   * Creates a new default cipher using 'UTF-8' encoding and key.
+   * 
+   * @param encryptionKey  A symmetric key to be used during encryption and decryption.
+   */
+  public Default1Cipher(byte[] encryptionKey) throws GeneralSecurityException {
+    this(encryptionKey, StandardCharsets.UTF_8);
+  }
+
+  /**
+   * Creates a new default cipher with the specified charset encoding, with a base64-encoded key.
+   * 
+   * @param base64CodedEncryptionKey  A base 64 encoded symmetric key to be used during encryption and decryption.
+   * @param charset                   The charset to be considered when encrypting plaintext or decrypting ciphertext.
+   */
+  public Default1Cipher(String base64CodedEncryptionKey, Charset charset) throws GeneralSecurityException {
+    this(Base64.getDecoder().decode(base64CodedEncryptionKey), charset);
+  }
+
+  /**
+   * Creates a new default cipher with the specified charset encoding and key.
+   * 
+   * @param encryptionKey  A symmetric key to be used during encryption and decryption.
+   * @param charset        The charset to be considered when encrypting plaintext or decrypting ciphertext.
+   */
+  public Default1Cipher(byte[] encryptionKey, Charset charset) throws GeneralSecurityException {
+    if (charset == null) throw new NullPointerException("charset");
+    if (encryptionKey == null) throw new NullPointerException("encryptionKey");
+    this.charset = charset;
+    this.key = new SecretKeySpec(encryptionKey, ENCRYPTION_ALGORITHM);
+  }
+
+  /** @{inheritDoc} */
+  @Override
+  public String encrypt(String plainText) {
+    try {
+      byte[] iv = new byte[IV_SIZE];
+      rnd.nextBytes(iv);
+      Cipher enc = Cipher.getInstance(CIPHER_ALGORITHM, PROVIDER);
+      enc.init(Cipher.ENCRYPT_MODE, key, new GCMParameterSpec(IV_SIZE * 8, iv));
```

Comment:
The first parameter to GCMParameterSpec should be the tag length, not the IV length. Tag length should be 128 bits